docs/docs/porting-to-vial.md

```diff
@@ -14,37 +14,78 @@ redirect_from:
+     If the `default` keyboard doesn't build, nothing based on it will either! Code copy-and-pasted in from other repositories may need to be modified [based on the version of QMK it was created for](https://docs.qmk.fm/#/breaking_changes?id=what-has-been-included-in-past-breaking-changes).
+
+> Note
+> {: .label .label-green }
+> Running `make` is not quite the same as running `qmk compile`. Running `qmk compile` in the wrong directory will default to compiling QMK firmware, not Vial firmware. To avoid any potential confusion, use `make`.
 
 
 ## 2. Create a new `vial` keymap
 
-Copy the existing `keymaps/default` keymap to `keymaps/vial`.
+Copy the existing `keymaps/default` folder to `keymaps/vial` with the content intact. Leave the `keymaps/default` folder in place; do not modify or remove it.
+
+> Important
+> {: .label .label-red }When porting to Vial, the resulting keymap should ALWAYS be named `vial` and nothing else! This avoids any confusion about if the keyboard has already been ported or not.
 
 ## 3. Enable Vial in your rules file
 
-Create a new file under `[keyboard_name]/keymaps/vial/rules.mk` with the following contents:<sup>[(example)](https://github.com/vial-kb/vial-qmk/blob/90f3b0e2e188eccb23ed8a2a690df278a0f1057b/keyboards/vial_example/vial_atmega32u4/keymaps/vial/rules.mk#L2)</sup>
+Create a new file (or edit an existing one) under `[keyboard_name]/keymaps/vial/rules.mk` with the following contents:<sup>[(example)](https://github.com/vial-kb/vial-qmk/blob/90f3b0e2e188eccb23ed8a2a690df278a0f1057b/keyboards/vial_example/vial_atmega32u4/keymaps/vial/rules.mk#L2)</sup>
 
 ```
 VIA_ENABLE = yes
 VIAL_ENABLE = yes
 ```
+> Important
+> {: .label .label-red }
+> Both of these rules are required. Do not remove `VIA_ENABLE`.
+
+Including these in a keyboard level `rules.mk` is not adviced.
 
 ## 4. Move JSON so Vial can find it
 
 Place your keyboard definition JSON (either one made in step 1 of this tutorial or downloaded from the [VIA keyboards repository](https://github.com/the-via/keyboards/tree/master/src)) under `[keyboard_name]/keymaps/vial/vial.json` so that Vial build process can find it. <sup>[(example)](https://github.com/vial-kb/vial-qmk/blob/90f3b0e2e188eccb23ed8a2a690df278a0f1057b/keyboards/vial_example/vial_atmega32u4/keymaps/vial/vial.json)</sup>
 
+> Important
+> {: .label .label-red }
+> This file must be named `vial.json`. It cannot have any other name.
+>
+> The Vial keyboard definition (`vial.json`) is not the same as an `info.json` file. You cannot copy an `info.json` file to use as the `vial.json`.
+
 ## 5. Generate and add unique keyboard ID
+The ID generated is unique for the firmware for the keyboard type and model; it does not need to be unique for each individual keyboard.
```

Comment:
not sure what this means? it should be unique for a specific pcb firmware, i.e. different pcb that uses different firmware (including rev1/rev2) should have different unique id
basically, if you cannot flash the same firmware to two pcbs, they should have different unique id

Reply:
Well, I made an attempt to clarify, dunno if it helps? It was an issue with more than one inexperienced user trying to create an unique ID for each keyboard of the same type.